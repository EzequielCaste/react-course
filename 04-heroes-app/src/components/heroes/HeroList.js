import React from 'react'
import { getHeroesByPublisher } from '../../selectors/getHeroesByPublisher'
import { HeroCard } from './HeroCard';


export const HeroList = ({ publisher }) => {
  const heroes = getHeroesByPublisher( publisher );
  //console.log(heroes);
  return (
    <div className="card-columns">
      <ul>
        {
          heroes.map( hero => {
            return <HeroCard 
            key={ hero.id }
            {...hero }
            />           
            
          })
        }
      </ul>
    </div>
  )
}
