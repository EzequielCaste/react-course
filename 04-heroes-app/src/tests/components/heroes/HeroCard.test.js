import React from 'react'

import { shallow } from 'enzyme'
import { HeroCard } from '../../../components/heroes/HeroCard'
import '@testing-library/jest-dom'

describe('Pruebas en <HeroCard />', () => {

  const wrapper = shallow(<HeroCard />);

  test('debe mostrarse correctamente', () => {

    expect( wrapper ).toMatchSnapshot();

  })
})