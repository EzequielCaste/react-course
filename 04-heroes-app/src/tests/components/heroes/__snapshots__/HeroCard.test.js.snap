// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <HeroCard /> debe mostrarse correctamente 1`] = `
<div
  className="card ms-3"
  style={
    Object {
      "maxWidth": 540,
    }
  }
>
  <div
    className="row no-gutters"
  >
    <div
      className="col-md-4"
    >
      <img
        className="card-img"
        src="./assets/heroes/undefined.jpg"
      />
    </div>
    <div
      className="col-md-8"
    >
      <div
        className="card-body"
      >
        <h5
          className="card-title"
        />
        <p
          className="card-text"
        />
        <p
          className="card-text"
        />
        <Link
          to="./hero/undefined"
        >
          MÃ¡s...
        </Link>
      </div>
    </div>
  </div>
</div>
`;
